src/hyp3_sdk/hyp3.py

```diff
@@ -40,17 +41,29 @@ def __init__(
                 Both username and password must be provided if either is provided.
             password: Password for authenticating to `urs.earthdata.nasa.gov`.
                 Both username and password must be provided if either is provided.
-            prompt: Prompt for username and/or password interactively when they
-                are not provided as keyword parameters
+            token: Earth Data Login Bearer Token for authenticating to `urs.earthdata.nasa.gov`
+            prompt: Either 'password' or 'token' to prompt for EDL user name and password or EDL bearer token, respectively.
         """
         self.url = api_url
 
-        if username is None and prompt:
-            username = input('NASA Earthdata Login username: ')
-        if password is None and prompt:
-            password = getpass('NASA Earthdata Login password: ')
+        if prompt and prompt != 'password' and prompt != 'token':
```

Comment:
Maybe something like this?
```suggestion
        if prompt not in (True, 'password', 'token'):
            raise ValueError(f'Unexpected value {prompt} for `prompt`')

        if prompt is True:
```